waiter/src/waiter/util/http_utils.clj

```diff
@@ -286,3 +286,12 @@
   [user-agent-products request]
   (when-let [user-agent (some-> request (get-in [:headers "user-agent"]) str/lower-case)]
     (some #(str/includes? user-agent %) user-agent-products)))
+
+(defn assoc-response-headers
+  "Attaches provided headers into the response."
+  [response & kvs]
+  (loop [headers (get response :headers)
+         [k v & rem-hvs] kvs]
+    (if (str/blank? k)
+      (assoc response :headers headers)
+      (recur (assoc headers k (str v)) rem-hvs))))
```

Comment:
Whenever you're tempted to use `loop` in Clojure, there's almost always a more idiomatic/declarative/functional way to do it instead. Unless you're implementing something really low-level — e.g., a new type of collection transformation — then I think it's always preferable to use higher-level functions instead.

```suggestion
(defn assoc-response-headers
  "Attaches provided headers into the response.
   Headers with blank values are excluded."
  [response & kvs]
  (let [xf (comp (map vec)
                 (remove (fn [[k v]] (str/blank? v))))
        kv-pairs (partition 2 kvs)]
    (update response :headers into xf kv-pairs)))
```

Reply:
This lgtm, will rewrite.

Reply:
I totally misunderstood the blank thing. This should be much simpler:

```
(defn assoc-response-headers
  "Attaches provided headers into the response."
  [response & kvs]
  (let [kv-pairs (partition 2 kvs)]
    (update response :headers into (map vec) kv-pairs)))
```

Reply:
Rewritten to use merge.